### PR TITLE
TSDB: Do not allow index splits for time series indices

### PR DESCRIPTION
**Pull request by csoulios**
Index `_split` operation may have some issues with time series indices, as
the splits must respect and adjust the time boundaries.

This PR forbids users to execute the split operation on time series indices, 
until we revisit this approach in the future.

### CODE REVIEW

File: rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/tsdb/80_index_resize.yml [hunk 1]
```diff
@@ -82,11 +82,11 @@ setup:
 ---
 split:
   - skip:
-      version: all
-      reason: shard splitting doesn't work yet
-      features: "arbitrary_key"
+      version: " - 8.00.99"
+      reason: indesx-split check introduced in 8.1.0
```
Comment (nik9000) [inline rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/tsdb/80_index_resize.yml:86]: s/indesx-split/index-split/

File: server/src/main/java/org/elasticsearch/action/admin/indices/shrink/TransportResizeAction.java [hunk 2]
```diff
@@ -124,6 +125,16 @@ protected void masterOperation(
...
 
+        // Index splits are not allowed for time-series indices
+        if (resizeRequest.getResizeType() == ResizeType.SPLIT) {
+            try {
+                IndexRouting.fromIndexMetadata(sourceMetadata).checkIndexSplitAllowed();
+            } catch (IllegalArgumentException e) {
+                listener.onFailure(e);
+                return;
+            }
+        }
```
Comment (nik9000) [inline server/src/main/java/org/elasticsearch/action/admin/indices/shrink/TransportResizeAction.java:136]: It doesn't look like the cleanest way to do it but it makes sense given what we have. And I don't think its permanent.

**Review by nik9000 — APPROVED** [review]

**Comment by csoulios** [comment]
@elasticmachine merge upstream

**Comment by csoulios** [comment]
@elasticmachine run elasticsearch-ci/part-2

**Pull request merged**